### PR TITLE
Refactor the 'cookie' property of the 'Code instrumentation' subsystem

### PR DESCRIPTION
**Pull request by LavrovArtem**
part of #1379

### CODE REVIEW

File: src/client/sandbox/native-methods.js [hunk 1]
```diff
@@ -39,13 +41,17 @@ class NativeMethods {
         this.documentCreateTouchList     = doc.createTouchList || docProto.createTouchList;
 
         // getters/setters
-        const docPrototype = win.Document.prototype;
+        const docPrototype             = win.Document.prototype;
+        const htmlDocPrototype         = win.HTMLDocument.prototype;
+        const documentCookieDescriptor = win.Object.getOwnPropertyDescriptor(isFirefox ? htmlDocPrototype : docPrototype, 'cookie');
```
Comment (miherlosev) [inline src/client/sandbox/native-methods.js]: ```this.documentCookiePropOwner = docPrototype.hasOwnProperty('cookie') ? 'Document' : 'HTMLDocument'```

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

File: src/client/sandbox/native-methods.js [hunk 2]
```diff
@@ -41,11 +39,12 @@ class NativeMethods {
         this.documentCreateTouchList     = doc.createTouchList || docProto.createTouchList;
 
         // getters/setters
-        const docPrototype             = win.Document.prototype;
-        const htmlDocPrototype         = win.HTMLDocument.prototype;
-        const documentCookieDescriptor = win.Object.getOwnPropertyDescriptor(isFirefox ? htmlDocPrototype : docPrototype, 'cookie');
+        const docPrototype = win.Document.prototype;
+
+        this.documentAllPropOwnerName    = docPrototype.hasOwnProperty('all') ? 'Document' : 'HTMLDocument';
+        this.documentCookiePropOwnerName = docPrototype.hasOwnProperty('cookie') ? 'Document' : 'HTMLDocument';
```
Comment (miherlosev) [inline src/client/sandbox/native-methods.js]: Move this code `docPrototype.hasOwnProperty('all') ? 'Document' : 'HTMLDocument';` 
to the instance level utils function `NativeMethods.getPropOwnerName (prop).
Store `docProtorype` in the `nativeMethods` if it's necessary.

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

File: src/client/sandbox/node/document/index.js [hunk 3]
```diff
@@ -235,5 +236,12 @@ export default class DocumentSandbox extends SandboxBase {
             },
             setter: null
         });
+
+        overrideDescriptor(window[nativeMethods.documentCookiePropOwnerName].prototype, 'cookie', {
```
Comment (miherlosev) [inline src/client/sandbox/node/document/index.js]: It's difficult to read. Let's move the prototype condition to separate variable:
```js
const documentCookiePropOwnerPrototype = window[nativeMethods.documentCookiePropOwnerName].prototype;

overrideDescriptor(documentCookiePropOwnerPrototype, 'cookie', {...});
```

Do this for `cookie` and `all` properties.

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Pull request merged**